scripts/common/addon.sh

```diff
@@ -231,3 +231,49 @@ function addon_set_has_been_applied() {
         kubectl patch configmaps -n kurl kurl-current-config --type merge -p "{\"data\":{\"addons-$name\":\"$current\"}}"
     fi
 }
+
+# check if the files are already present - if they are, use that
+# if they are not, prompt the user to provide them
+# if the user does not provide the files, return 1
+function addon_fetch_airgap() {
+    local name=$1
+    local version=$2
+    local package="$name-$version.tar.gz"
+
+    if [ -f "$(package_filepath "${package}")" ]; then
+        # the package already exists, no need to download it
+        printf "The package %s %s is already available locally.\n" "$name" "$version"
+    else
+        # prompt the user to give us the package
+        printf "The package %s %s is not available locally, and is required.\n" "$name" "$version"
+        printf "You can download it from %s with the following command:\n" "$(get_dist_url)/${package}"
+        printf "\n${GREEN}    curl -o %s %s${NC}\n\n" "${package}" "$(get_dist_url)/${package}"
+
+        if ! prompts_can_prompt; then
+            # we can't ask the user to give us the file because there are no prompts, but we can say where to put it for a future run
+            printf "Please move this file to %s before rerunning the installer.\n" "$(package_filepath "${package}")"
+            return 0
+        fi
+
+        printf "If you have this file, please provide the path to the file on the server.\n"
+        printf "If you do not have the file, leave the prompt empty and this package will be skipped.\n"
+        printf "%s %s filepath: " "$name" "$version"
+        prompt
+        if [ -n "$PROMPT_RESULT" ]; then
+            local loadedPackagePath="$PROMPT_RESULT"
+            cp "$loadedPackagePath" "$(package_filepath "${package}")"
+        else
+            printf "Skipping package %s %s\n" "$name" "$version"
+            printf "You can provide the path to this file the next time the installer is run,"
+            printf "or move it to %s to be detected automatically.\n" "$(package_filepath "${package}")"
+            return 1
+        fi
+    fi
+
+    printf "Unpacking %s %s...\n" "$name" "$version"
+    tar xf "$(package_filepath "${package}")"
```

Comment:
does tar automatically detect gzip compression?

Reply:
Good question
I copied this line from elsewhere though, and it worked in my test

Reply:
yeah, tar automatically decompresses: https://unix.stackexchange.com/questions/253596/tar-extraction-also-automatically-decompresses